_source/_posts/2023-09-15-java-spring-boot-microservices.md

```diff
@@ -0,0 +1,722 @@
+
+```shell
+git clone https://github.com/oktadev/auth0-java-microservices-examples.git
+cd auth0-java-microservices-examples/spring-boot-gateway-webflux
+```
+
+In the `spring-boot-gateway-webflux` directory, there are three projects:
+
+* **discovery-service**: a Netflix Eureka server used for service discovery.
+* **car-service**: a simple Car Service that uses Spring Data REST to serve up a REST API of cars.
+* **api-gateway**: an API gateway with a `/cool-cars` endpoint that talks to the car service and filters out cars that aren't cool (in my opinion, of course).
+
+### Run a Secure Spring Boot Microservice Architecture
+
+To run the example, you must [install the Auth0 CLI](https://github.com/auth0/auth0-cli#installation) and create an Auth0 account. If you don't have an Auth0 account, [sign up for free](https://auth0.com/signup). I recommend using [SDKMAN!](https://sdkman.io) to install Java 17+.
+
+First, start the discovery service:
+
+```shell
+cd discovery-service
+./gradlew bootRun
+```
+
+Before it starts, you'll need to configure the API gateway to use your Auth0 account.
+
+Open a terminal and run `auth0 login` to configure the Auth0 CLI to get an API key for your tenant. Then, run `auth0 apps create` to register an OpenID Connect (OIDC) app with the appropriate URLs:
+
+```shell
+auth0 apps create \
+  --name "Kick-Ass Cars" \
+  --description "Microservices for Cool Cars" \
+  --type regular \
+  --callbacks http://localhost:8080/login/oauth2/code/okta \
+  --logout-urls http://localhost:8080 \
+  --reveal-secrets
+```
+
+Copy `api-gateway/.env.example` to `.env` and edit it to contain the values from the command above.
+
+```dotenv
+OKTA_OAUTH2_ISSUER=https://<your-auth0-domain>/
+OKTA_OAUTH2_CLIENT_ID=
+OKTA_OAUTH2_CLIENT_SECRET=
+OKTA_OAUTH2_AUDIENCE=https://<your-auth0-domain>/api/v2/
+```
+
+At startup, these properties will be read using [spring-dotenv](https://github.com/paulschwarz/spring-dotenv).
+
+Run `./gradlew bootRun` to start the API gateway, or use your IDE to run it.
+
+Copy `car-service/.env.example` to `.env` and update its values.
+
+```dotenv
+OKTA_OAUTH2_ISSUER=https://<your-auth0-domain>/
+OKTA_OAUTH2_AUDIENCE=https://<your-auth0-domain>/api/v2/
+```
+
+Start it with `./gradlew bootRun` and open `http://localhost:8080` in your favorite browser. You'll be redirected to Auth0 to log in.
+
+{% img blog/spring-boot-microservices/auth0-login.png alt:"Auth0 Login" %}{: .center-image }
+
+After authenticating, you'll see your name in lights! ✨
+
+{% img blog/spring-boot-microservices/name-in-lights.png alt:"Your name in lights" %}{: .center-image }
+
+You can navigate to the following URLs in your browser for different results:
+
+- `http://localhost:8080/print-token`: prints access token to the console
+- `http://localhost:8080/cool-cars`: returns a list of cool cars
+- `http://localhost:8080/home`: proxies request to the car service and prints JWT claims to the console
```

Comment:
I got confused on where to look. I might say "terminal" instead of console, as it was used before. Also for /home, I would say it prints in the car-service terminal, I was looking at the api-gateway terminal.

Reply:
Please let me know if the suggested change works for you.

Reply:
It works